### ISSUE TITLE
used repeated field instead of map

### ISSUE THREAD

**Issue by qiwzhang** [istio/api#108]
It is better to use repeated field.  map is very slow to build even after we fixes random seed issue.  
In a couple gdb sections in istio/proxy,  I notice that they are in map destructor.

If we use repeated field for our attributes,  we can have one repeated field for all attribute types and use "oneof" for the value type. 

it will be like:

repeated Attribute  attributes = 3

message Attribute {
      int   name_index = 1;
      one of {
            int64  int64 = 1;
            double double = 2;
            int   string_index = 3;
...
     }
}

@geeknoid what do you think?

**Comment by geeknoid** [comment]
I actually implemented this a few weeks ago, cause I also realized that
we're wasting time creating maps for this stuff on the Mixer. I abandoned
the effort 1/2 through however because I was having some problems with Go
and the oneofs. In particular, if I recall correctly, the issue was that
every value ended up being boxed as a separate allocation in the heap. So
even if you send an int32, the use in the oneof could cause the int32 to be
hoisted into the heap as a separate object. Any gains you got by not having
a map gets lost by substantially increasing the load on the GC.

So I gave up and moved on. It's still an interesting idea which I'm
interested to returning to.
